openvpn/tunnel/setup_tun_linux.go

```diff
@@ -113,6 +121,24 @@ func (service *LinuxTunDeviceManager) deviceExists(device tunDevice) (exists boo
 	return false, err
 }
 
+func (service *LinuxTunDeviceManager) deviceUsed(device tunDevice) (exists bool, err error) {
+	contents, err := ioutil.ReadFile("/sys/class/net/" + device.Name + "/carrier")
+	if err != nil {
+		return false, err
+	}
+
+	value, err := strconv.Atoi(string(contents))
+	if err != nil {
+		return false, err
+	}
+
+	if value == 1 {
```

Comment:
This can be a bit smaller with just return:
```suggestion
	return value == 1, nil
```